packages/pages/src/components/Image/Image.tsx

```diff
@@ -0,0 +1,188 @@
+   */
+  placeholder?: string;
+}
+
+export interface ImagePropsWithSrc extends ImagePropsShared {
+  src: string;
+  imageField?: never;
+}
+
+export interface ImagePropsWithField extends ImagePropsShared {
+  src?: never;
+  imageField: ImageType;
+}
+
+export type ImageProps = ImagePropsWithSrc | ImagePropsWithField;
+
+/**
+ * The 'Image' component is used to render an HTML <image> element
+ *
+ * @example
+ * ```
+ * <Image src="https://via.placeholder.com/400x600" />
+ * <Image imageField={profile.c_heroImage} />
+ * <Image imageField={profile.c_heroImage} bpSizes={['400x, '600x400', 'x800']} />
+ * ```
+ *
+ * @param {ImageType} imageField
+ *          Yext Image field, used to generate `src` (and `srcSet`, `sizes` if [bpSizes | customSizes] are specified).
+ * @param {[string, string, string]} bpSizes
+ *          Size of requested image at breakpoints [sm, md, lg]. Size in pixels should be unitless in the format `[width]x[height]` ex. `400x600` `400x` `x600`
+ * @param {[[string, string]]} customSizes
+ *          Expects format `[[bp, size], ['600px', '300x400'], ...]`.  Like bpSizes, but you can define the breakpoints yourself.
+ * @param {string} src
+ *          HTML attribute. This overrides the generated value from imageField.
+ * @param {string} srcSet
+ *          HTML attribute. This overrides the generated value from [bpSizes | customSizes].
+ * @param {string} sizes
+ *          HTML attribute. This overrides the generated value from [bpSizes | customSizes].
+ * @param {string} alt
+ *          HTML attribute. Defaults to empty string
+ * @param {string} loading
+ *          HTML attribute. Defaults to 'lazy'
+ * @param {string} layout
+ *          The size of the image and its resizing behavior. "fixed" | "constrained" | "full-width" are supported. Defaults to 'contrained'
+ * @param {string} aspectRatio
+ *          Should be 'height / width'.
+ *          Some common aspect ratios:
+ *            1:1 = 1.0, 4:3 = 0.75, 16:9 = 0.5625
+ * @param {number} width
+ * @param {number} height
+ * @param {number} maxWidth
+ * @param {string} objectFit
+ *          CSS property sets how the content of a replaced element. Defaults to "cover"
+ * @param {string} objectPosition
+ *          CSS property specifies the alignment of the selected replaced element's contents within the element's box. Defaults to "center"
+ * @param {string} placeholder
+ *
+ * @returns {Image}
+ */
+const Image = React.forwardRef<HTMLImageElement, ImageProps>((props, ref) => {
+  const {
+    imageField,
+    bpSizes,
+    customSizes,
+    layout = "contrained",
+    aspectRatio,
+    objectFit = "cover",
+    objectPosition = "center",
+    width,
+    height,
+    maxWidth,
+    placeholder,
+    src,
+    srcSet,
+    sizes,
+    alt = "",
+    loading = "lazy",
+    ...rest
+  } = props;
+
+  // One of `src`, `imageField` is required to be populated in props, so manually
+  //    casting imageField as `imageType` here to ignore typescript type checking
```

Comment:
weird spacing. Can you run `pnpm fmt`? Though the auto formatter should have run. Maybe it's not checking in this directory for some reason? Though I think it should be.

Reply:
It looks like intentional reverse-indentation to indicate it's part of the same comment paragraph.